addons/mail/models/discuss/discuss_channel_member.py

```diff
@@ -77,6 +78,20 @@ def _search_is_self(self, operator, operand):
                 ("guest_id", "!=", current_guest.id) if current_guest else expression.TRUE_LEAF,
             ]
 
+    def _search_is_pinned(self, operator, operand):
+        if (operator == "=" and operand) or (operator == "!=" and not operand):
+            return [
+                '|', '|',
+                ('unpin_dt', '=', False), ('last_interest_dt', '>', self._field_to_sql(self._table, 'unpin_dt')),
+                ('channel_id.last_interest_dt', '>', self._field_to_sql(self._table, 'unpin_dt')),
```

Comment:
I'm afraid this could be a disaster for performance as it will require linking another table every time.
This should be benchmarked.

Reply:
I am also worried about that so at first, I just wanted to introduce `unpin_dt` and get rid of `is_pinned` totally but then I found there is a lot of searching on `is_pinned` in the code and we have a view that includes the `is_pinned` so it can be adjusted in the view page. If it is gone then it will be a functional change. So I kept the `is_pinned` and only made it a compute field. But seems indeed this can lead to some performance issues as well. Do you have a better idea?

Reply:
Just benchmark it (start with populate medium), maybe it's fine.

Reply:
i dont see a lot of difference from master, 251ms vs 255ms for `_channel_info`, which is the biggest bottleneck calling the sql search containing the change.